src/Util/Image.cc

```diff
@@ -21,3 +21,29 @@ CARTA::PolarizationType GetStokesType(int stokes_value) {
     }
     return stokes_type;
 }
+
+void GetTiles(int image_width, int image_height, int mip, std::vector<carta::Tile>& tiles) {
+    int tile_size_original = TILE_SIZE * mip;
+    int num_tile_columns = ceil((double)image_width / tile_size_original);
+    int num_tile_rows = ceil((double)image_height / tile_size_original);
+    int32_t tile_layer = carta::Tile::MipToLayer(mip, image_width, image_height, TILE_SIZE, TILE_SIZE);
+    tiles.resize(num_tile_rows * num_tile_columns);
+
+    for (int i = 0; i < num_tile_columns; ++i) {
+        for (int j = 0; j < num_tile_rows; ++j) {
+            tiles[j * num_tile_columns + i].x = i;
+            tiles[j * num_tile_columns + i].y = j;
+            tiles[j * num_tile_columns + i].layer = tile_layer;
+        }
+    }
+}
```

Comment:
This is only used in Frame in one single place. Do you really need a different function somewhere else? If this is not a common utility function, I suggest you move it to Frame (or maybe Tile.h?). You are actually passing two parameters which are members of Frame.

Also, this is used only in `Frame::VectorFieldImage`, which is already a very long method. There is no harm with just placing it there unless you want to test this function in isolation, for example.